wandb/integration/keras/keras.py

```diff
@@ -642,58 +704,19 @@ def _log_weights(self):
         return metrics
 
     def _log_gradients(self):
-        if not self.training_data:
-            raise ValueError("Need to pass in training data if logging gradients")
-
-        X_train = self.training_data[0]
-        y_train = self.training_data[1]
+        weights = self.model.trainable_weights
+        grads = [np.zeros(tuple(w.shape)) for w in weights]
+        for x, y in self._training_data_generator():
+            with tf.GradientTape() as tape:
```

Comment:
Awesome, how does this handle tf1.x?  can we make sure it fails (or doesnt log gradients) gracefully?